nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/data/explorer/PutAzureDataExplorer.java

```diff
@@ -0,0 +1,309 @@
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .defaultValue("false")
+            .build();
+    public static final PropertyDescriptor ADX_SERVICE = new PropertyDescriptor
+            .Builder().name("Kusto Ingest Service")
+            .displayName("Kusto Ingest Service")
+            .description("Azure Data Explorer Kusto Ingest Service")
+            .required(true)
+            .identifiesControllerService(KustoIngestService.class)
+            .build();
+
+    static final PropertyDescriptor DATA_FORMAT = new PropertyDescriptor.Builder()
+            .name("Data Format")
+            .displayName("Data Format")
+            .description("The format of the data that is sent to Azure Data Explorer.")
+            .required(true)
+            .allowableValues(KustoIngestDataFormat.values())
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor IGNORE_FIRST_RECORD = new PropertyDescriptor.Builder()
+            .name("Ingestion Ignore First Record")
+            .displayName("Ingestion Ignore First Record")
+            .description("Defines whether ignore first record while ingestion.")
+            .required(false)
+            .allowableValues(IGNORE_FIRST_RECORD_YES, IGNORE_FIRST_RECORD_NO)
+            .defaultValue(IGNORE_FIRST_RECORD_NO.getValue())
+            .build();
+
+    static final PropertyDescriptor ROUTE_PARTIALLY_SUCCESSFUL_INGESTION = new PropertyDescriptor.Builder()
+            .name("Route partially successful ingestion records")
+            .displayName("Route partially successful ingestion records")
+            .description("Defines where to route partially successful ingestion records.")
+            .required(false)
+            .allowableValues("Success", "Failure")
+            .defaultValue("Failure")
+            .build();
+
+    static final PropertyDescriptor INGESTION_STATUS_POLLING_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("Timeout for polling on ingestion status")
+            .displayName("Timeout for polling on ingestion status in seconds")
+            .description("Defines the value of timeout for polling on ingestion status in seconds")
+            .required(false)
+            .dependsOn(POLL_ON_INGESTION_STATUS, "true")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("600")
+            .build();
+
+    static final PropertyDescriptor INGESTION_STATUS_POLLING_INTERVAL = new PropertyDescriptor.Builder()
+            .name("Ingestion status polling interval")
+            .displayName("Ingestion status polling interval in seconds")
+            .description("Defines the value of timeout for polling on ingestion status in seconds.")
+            .required(false)
+            .dependsOn(POLL_ON_INGESTION_STATUS, "true")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("5")
```

Comment:
See above for recommendation on using the Time Period validator.
```suggestion
            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
            .defaultValue("5 s")
```